application/src/main/java/org/togetherjava/tjbot/commands/modmail/ModmailCommand.java

```diff
@@ -0,0 +1,138 @@
+import net.dv8tion.jda.api.events.interaction.SlashCommandEvent;
+import net.dv8tion.jda.api.interactions.commands.OptionType;
+import net.dv8tion.jda.api.interactions.components.ActionRow;
+import net.dv8tion.jda.api.interactions.components.selections.SelectOption;
+import net.dv8tion.jda.api.interactions.components.selections.SelectionMenu;
+import org.jetbrains.annotations.NotNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.togetherjava.tjbot.commands.SlashCommandAdapter;
+import org.togetherjava.tjbot.commands.SlashCommandVisibility;
+import org.togetherjava.tjbot.config.Config;
+
+import java.util.*;
+
+public class ModmailCommand extends SlashCommandAdapter {
+    private static final Logger logger = LoggerFactory.getLogger(ModmailCommand.class);
+
+    private static final String COMMAND_NAME = "modmail";
+    private static final String TARGET_OPTION = "message";
+    private static final Config config = Config.getInstance();
+
+    static final List<SelectOption> mods = new ArrayList<>();
+    static final Map<String, User> modsMap = new HashMap<>();
+
+    private final JDA jda;
+
+    /**
+     * Creates an instance of the ModMail command.
+     *
+     * @param jda the {@link JDA} instance of all slash commands to find the target guild or the
+     *        guild where the moderators are.
+     */
+    public ModmailCommand(@NotNull JDA jda) {
+        super(COMMAND_NAME,
+                "sends a message to either a single moderator or on the mod_audit_log channel",
+                SlashCommandVisibility.GLOBAL);
+
+        getData().addOption(OptionType.STRING, TARGET_OPTION, "The message to send", true);
+
+        this.jda = Objects.requireNonNull(jda);
+
+        mods.add(SelectOption.of("All Moderators", "all"));
+    }
+
+    @Override
+    public void onSlashCommand(@NotNull SlashCommandEvent event) {
+        String memberId = event.getUser().getId();
+
+        // checks if selection menu already contains the moderators
+        if (mods.size() == 1) {
+            loadMenuOptions();
+        }
+
+        event.reply("""
+                Select the moderator to send message to, or select "All Moderators" to send to
+                the guild's mod audit channel.
+                """)
+            .addActionRow(SelectionMenu
+                .create(generateComponentId(memberId, event.getOption(TARGET_OPTION).getAsString()))
+                .addOptions(mods)
+                .build())
+            .setEphemeral(false)
+            .queue();
+    }
+
+    @Override
+    public void onSelectionMenu(@NotNull SelectionMenuEvent event, @NotNull List<String> args) {
+        String message = args.get(1);
+        // Ignore if another user clicked the button which is only possible when used within the
+        // guild.
+        String userId = args.get(0);
+        if (event.isFromGuild()
+                && !userId.equals(Objects.requireNonNull(event.getMember()).getId())) {
```

Comment:
```suggestion
                && !userId.equals(event.getMember().getId())) {
```

You're effectively doing

```java
if (member == null) {
    member.toString();
}
```
Here, which is, useless